src/Build.UnitTests/FixPathOnUnix_Tests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Xunit;
+
+namespace Microsoft.Build.UnitTests
+{
+    [PlatformSpecific(Xunit.PlatformID.AnyUnix)]
+    public  class FixPathOnUnixTests
+    {
+        [Fact]
+        public void TestPathFixupInMetadata()
+        {
+            string buildProjectContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                    <Target Name='Build'>
+                        <MSBuild Projects='projectDirectory/main.proj' />
+                    </Target>
+               </Project>";
+
+            string mainProjectContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                    <UsingTask TaskName='LogTaskPropertiesTask' AssemblyName='Microsoft.Build.Engine.UnitTests' />
+                    <ItemGroup>
+                        <Item0 Include='xyz'>
+                            <Md0>lib\foo.dll</Md0>
+                        </Item0>
+                    </ItemGroup>
+                    <Target Name='Build'>
+                        <LogTaskPropertiesTask Items='@(Item0)' />
+                    </Target>
+                </Project>";
+
+            string buildProjectPath = ObjectModelHelpers.CreateFileInTempProjectDirectory("build.proj", buildProjectContents);
+            string mainProjectPath = ObjectModelHelpers.CreateFileInTempProjectDirectory("projectDirectory/main.proj", mainProjectContents);
+            ObjectModelHelpers.CreateFileInTempProjectDirectory(Path.Combine("projectDirectory", "lib", "foo.dll"), "just a text file");
+
+            var projColln = new ProjectCollection();
```

Comment:
It's sad we don't have a superclass for all our tests, but:

- `ProjectCollection` is disposable, please put it in a using block
- Pass a newly created `ProjectRootElementCache` to the `ProjectCollection` constructor, and then put the cache in the using block as well. Without this step, the default cache is used, which may then interact with other tests miss-using the global xml cache.